planner/core/rule_partition_processor.go

```diff
@@ -762,11 +762,34 @@ func (s *partitionProcessor) pruneRangePartition(ctx sessionctx.Context, pi *mod
 		monotonous: mono,
 	}
 	result = partitionRangeForCNFExpr(ctx, conds, &pruner, result)
+	// remove useless predicates after partition pruning
+	if condsToBePruned != nil {
```

Comment:
We could change the codes here to
```
if condsToBePruned == nil {
	return result, nil
}
large code block.
return result, nil
```

This is a common coding style in TiDB

Reply:
Sure